.github/scripts/test_entrypoint.sh

```diff
@@ -0,0 +1,301 @@
+		fi
+
+		# When a library has a SONAME, there should be a symlink with the SONAME
+		# pointing to the library file. This is usually in the same directory.
+		local lib_dir
+		lib_dir=$(dirname "$file")
+		if [ ! -L "$lib_dir/$soname" ]; then
+			status_fail "Library $file has SONAME '$soname' but no corresponding symlink was found in $lib_dir"
+			has_failure=1
+		elif [ "$(readlink -f "$lib_dir/$soname")" != "$(readlink -f "$file")" ]; then
+			status_fail "Symlink for SONAME '$soname' does not point to $file"
+			has_failure=1
+		else
+			status_pass "SONAME link for $file is correct"
+		fi
+	else
+		status_warn "Library $file doesn't have a SONAME"
+	fi
+
+	if [ "$has_failure" = 1 ]; then
+		return 1
+	fi
+
+	return 0
+}
+
+do_generic_tests() {
+	local all_files
+	if is_opkg; then
+		all_files=$(opkg files "$PKG_NAME")
+	elif is_apk; then
+		all_files=$(apk info --contents "$PKG_NAME" | sed 's#^#/#')
+	fi
+
+	local files
+	files=$(echo "$all_files" | grep -E '^(/bin/|/sbin/|/usr/bin/|/usr/libexec/|/usr/sbin/|/lib/|/usr/lib/)')
+
+	local has_failure=0
+	for file in $files; do
+		if [ ! -e "$file" ]; then
+			# opkg files can list directories
+			continue
+		fi
+
+		# Check if it is a symlink and if the target exists
+		if [ -L "$file" ]; then
+			if [ -e "$(readlink -f "$file")" ]; then
+				status_pass "Symlink $file points to an existing file"
+			else
+				status_fail "Symlink $file points to a non-existent file"
+				has_failure=1
+			fi
+
+			# Skip symlinks
+			continue
+		fi
+
+		if is_exec "$file" && ! check_exec "$file"; then
+			has_failure=1
+		fi
+
+		# Skip non-ELF files
+		if ! file "$file" | grep -q "ELF"; then
+			continue
+		fi
+
+		check_hardcoded_paths "$file"
+
+		if file "$file" | grep 'not stripped'; then
+			status_warn "Binary $file is not stripped"
+		else
+			status_pass "Binary $file is stripped"
+		fi
+
+		if ! check_linked_libs "$file"; then
+			has_failure=1
+		fi
+
+		if is_lib "$file" && ! check_lib "$file"; then
+			has_failure=1
+		fi
+	done
+
+	if [ "$has_failure" = 1 ]; then
+		err "Generic tests failed"
+		return 1
+	fi
+
+	success "Generic tests passed"
+	return 0
+}
+
+if is_opkg; then
+	echo "src/gz packages_ci file:///ci" >> /etc/opkg/distfeeds.conf
+	# Disable checking signature for all opkg feeds, since it doesn't look like
+	# it's possible to do it for the local feed only, which has signing removed.
+	# This fixes running CI tests.
+	sed -i '/check_signature/d' /etc/opkg.conf
+	opkg update
+	opkg install binutils file
+elif is_apk; then
+	echo "/ci/packages.adb" >> /etc/apk/repositories.d/distfeeds.list
+	apk update
+	apk add binutils file
+fi
+
+if generic_tests_enabled && generic_tests_forced; then
+	warn 'Generic tests are enabled and forced'
+elif generic_tests_enabled; then
+	warn 'Generic tests are enabled'
+else
+	warn 'Generic tests are disabled'
+fi
+
+for PKG in /ci/*.[ai]pk; do
+	if is_opkg; then
+		tar -xzOf "$PKG" ./control.tar.gz | tar xzf - ./control
```

Comment:
What if tar fails to extract it?

Reply:
Oh... yeah, it is quite confusing. :-/ Would be easier to review things, which we can see what you modified/added/changed, but I will prepare PR against your repo.